threaded_io_test.cc

```diff
@@ -51,6 +57,11 @@ int main(int argc, char* argv[]) {
     return 1;
   }
 
+#ifdef H5_TIMING_ENABLE
```

Comment:
Again, please remove all changes you made to this function and put the needed calls into HDFOutputer as that is the code which requires this setup and teardown. Especially since we need to be able to build threaded_io_test.cc even when we have no HDF libraries.